x-pack/filebeat/input/cometd/input_test.go

```diff
@@ -31,266 +33,295 @@ const (
-				eventsCh <- event
-				return true
-			},
-		}
-		connector := &mockedConnector{
-			outlet: outlet,
-		}
+	const numMessages = 1
 
-		var expected bay.MaybeMsg
-		expected.Msg.Data.Event.ReplayID = 1234
-		expected.Msg.Data.Payload = []byte(`{"CountryIso": "IN"}`)
-		expected.Msg.Channel = firstChannel
-
-		config1 := map[string]interface{}{
-			"channel_name":              firstChannel,
-			"auth.oauth2.client.id":     "client.id",
-			"auth.oauth2.client.secret": "client.secret",
-			"auth.oauth2.user":          "user",
-			"auth.oauth2.password":      "password",
-		}
-		config2 := map[string]interface{}{
-			"channel_name":              secondChannel,
-			"auth.oauth2.client.id":     "client.id",
-			"auth.oauth2.client.secret": "client.secret",
-			"auth.oauth2.user":          "user",
-			"auth.oauth2.password":      "password",
-		}
+	var eventProcessing sync.WaitGroup
+	eventProcessing.Add(numMessages)
 
-		// create Server
-		r := http.HandlerFunc(oauth2Handler)
-		server := httptest.NewServer(r)
-		serverURL = server.URL
-		config1["auth.oauth2.token_url"] = serverURL + "/token"
-		config2["auth.oauth2.token_url"] = serverURL + "/token"
+	outlet := &mockedOutleter{
+		onEventHandler: func(event beat.Event) bool {
+			eventProcessing.Done()
+			eventsCh <- event
+			return true
+		},
+	}
+	connector := &mockedConnector{
+		outlet: outlet,
+	}
+	var inputContext finput.Context
+
+	var expected bay.MaybeMsg
+	expected.Msg.Data.Event.ReplayID = 1234
+	expected.Msg.Data.Payload = []byte(`{"CountryIso": "IN"}`)
+	expected.Msg.Channel = firstChannel
+
+	config := map[string]interface{}{
+		"channel_name":              firstChannel,
+		"auth.oauth2.client.id":     "client.id",
+		"auth.oauth2.client.secret": "client.secret",
+		"auth.oauth2.user":          "user",
+		"auth.oauth2.password":      "password",
+	}
 
-		// get common config
-		cfg1 := common.MustNewConfigFrom(config1)
-		cfg2 := common.MustNewConfigFrom(config2)
+	r := http.HandlerFunc(oauth2Handler)
+	server := httptest.NewServer(r)
+	defer server.Close()
 
-		var inputContext finput.Context
+	serverURL = server.URL
+	config["auth.oauth2.token_url"] = server.URL + "/token"
 
-		// initialize inputs
-		input1, err := NewInput(cfg1, connector, inputContext)
-		require.NoError(t, err)
-		require.NotNil(t, input1)
+	cfg := common.MustNewConfigFrom(config)
 
-		input2, err := NewInput(cfg2, connector, inputContext)
-		require.NoError(t, err)
-		require.NotNil(t, input2)
+	input, err := NewInput(cfg, connector, inputContext)
+	require.NoError(t, err)
+	require.NotNil(t, input)
 
-		// run input
-		input1.Run()
-		defer input1.Stop()
+	require.Equal(t, 0, bay.GetConnectedCount())
+	input.Run()
+	eventProcessing.Wait()
+	require.Equal(t, 1, bay.GetConnectedCount())
 
-		event1 := <-eventsCh
-		assertEventMatches(t, expected, event1)
+	go func() {
+		time.Sleep(100 * time.Millisecond) // let input.Stop() be executed.
+		for range eventsCh {
```

Comment:
There is nothing in this goroutine that could fail the test. What do you want it to do?

If you want to ensure it exits before the test does you would need to use a WaitGroup or similar to track that.

Reply:
Sure. l can use connection count to check how many open channel are there and based on that I can determine if input is closed or not. as we already know that `eventCh` is not useful for this particular scenario.